fenix/app/src/main/java/org/mozilla/fenix/translations/TranslationsDialogBinding.kt

```diff
@@ -0,0 +1,145 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.translations
+
+import android.content.Context
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.distinctUntilChangedBy
+import kotlinx.coroutines.flow.mapNotNull
+import mozilla.components.browser.state.selector.findTab
+import mozilla.components.browser.state.state.BrowserState
+import mozilla.components.browser.state.store.BrowserStore
+import mozilla.components.concept.engine.translate.initialFromLanguage
+import mozilla.components.concept.engine.translate.initialToLanguage
+import mozilla.components.lib.state.helpers.AbstractBinding
+import org.mozilla.fenix.R
+
+/**
+ * Helper for observing Translation state from browserStore.
+ */
+class TranslationsDialogBinding(
+    val browserStore: BrowserStore,
+    private val translationsDialogStore: TranslationsDialogStore,
+    private val sessionId: String,
+    private val context: Context,
```

Comment:
Let's remove this direct dependency on `Context`, as it is an anti-pattern to give classes access to a god-object like this. It implies that future additions to the class could have access to anything in the god object. Instead, you can compute the string using a lambda. Something like:

```
// constructor
private val getTranslatedPageTitle: (localizedFrom: String, localizedTo: String) -> String

// usage
translationsDialogStore.dispatch(
                        TranslationsDialogAction.UpdateTranslatedPageTitle(
                            getTranslatedPageTitle(from, to),
                        ),
)

// initialization
TranslationsDialogBinding(getTranslatedPageTitle = { from, to -> 
          context.context.getString(
                                R.string.translations_bottom_sheet_title_translation_completed,
                                from,
                                to
                            ),
)
```

sorry for the formatting there, in a bit of a rush today

Reply:
Done